third_party/terraform/resources/resource_dns_record_set.go

```diff
@@ -334,3 +345,76 @@ func ipv6AddressDiffSuppress(_, old, new string, _ *schema.ResourceData) bool {
 
 	return oldIp.Equal(newIp)
 }
+
+func resourceDnsRecordSetReadUpdate(d *schema.ResourceData, meta interface{}) error {
```

Comment:
Potentially `resourceDnsRecordSetUpdate` could be adjusted to feature this same logic (read from API, use that as old, update with new) to make the code more DRY-friendly, but wanted to get thoughts on this first. It'd make the PR smaller but could introduce other unforeseen bugs.

Reply:
Can't you use the same/similar logic we already have for NS records?

Reply:
@drebes Yes! Updated the PR.